### ISSUE TITLE
PQ2 FlowScript Compile Fails

### ISSUE THREAD

**Issue by AcidicPeridot** [tge-was-taken/Atlus-Script-Tools#76]
[This .bf file](https://github.com/tge-was-taken/Atlus-Script-Tools/files/15107046/combi01.zip) from Persona Q2 throws an error when attempting to recompile, even when no changes are made.

Command:
`"combi01.bf.flow" -Compile  -OutFormat V2 -Library PQ2 -Encoding SJ  -Hook  -SumBits
`
Error:
```
[04/25/2024 02:53 AM] 4/25/2024 2:53:58 AM  Info: (0103:0000) Compiling procedure: BTL_TAG_RUN
[04/25/2024 02:53 AM] 4/25/2024 2:53:58 AM  Error: (0107:0004) A integer literal is an invalid statement
[04/25/2024 02:53 AM] 4/25/2024 2:53:58 AM  Error: (0104:0000) Failed to emit procedure body
[04/25/2024 02:53 AM] 4/25/2024 2:53:58 AM  Error: One or more errors occured during compilation!
[04/25/2024 02:53 AM] 4/25/2024 2:53:58 AM  Error: One or more errors occured while executing task!
[04/25/2024 02:53 AM] 
```

**Comment by AnimatedSwine37** [comment]
The actual problem is with the decompilation. The procedure ends up as the following when is clearly incorrect and therefore can't be compiled

```c
void BTL_TAG_RUN()
{
    PUTS( "実行" );
    EVT_FUNCTION_0018( 4, 9001 );
    0;
    0;
    UNKNOWN_FUNCTION_12707();
    60;
    1;
    UNKNOWN_FUNCTION_12707();
    140;
    UNKNOWN_FUNCTION_12708();
    300;
    2;
    UNKNOWN_FUNCTION_12707();
    BTL_FUNCTION_0198();
}
```

**Comment by lraty-li** [comment]
same problem in event/e800 series .bf files. 
event/e800/e808_020.bf.flow    line 26
```
......
// Procedure Index: 0
void e808_020()
{
    int var0;
    EVT_FUNCTION_000F( 1 );
    EVT_FUNCTION_0001( -1, -1 );
    EVT_FUNCTION_0017( 1 );
    EVT_SET_USUAL();
    0;
    1;
    
    if ( UNKNOWN_FUNCTION_-32764() == 8 )
    {
        WND_FUNCTION_000C( 0, 1 );
        WND_FUNCTION_000B( 2 );
        WND_FUNCTION_000B( 3 );
......
```
and unknown function numer gets negative value,  like -32764 of UNKNOWN_FUNCTION_-32764()
<img src="https://github.com/tge-was-taken/Atlus-Script-Tools/assets/46740834/eab0562d-d924-4c0f-9248-6388d1493b99" height="350px">
[e828_045.zip](https://github.com/tge-was-taken/Atlus-Script-Tools/files/15146383/e828_045.zip)

**Comment by oceanstuck** [comment]
Went and did a look with this alongside Peridot. The initial example with `UNKNOWN_FUNCTION`s 12707 and 12708 was the result of the library having no functions at their respective ids (0x31a3 and 0x31a4) and was resolved by adding functions of those ids with the correct number of parameters. They appear to both control the displaying of subtitles during Unison Skills, with 12707 displaying a msg at a certain frame and 12708 controlling how long a msg is displayed for.

However, `UNKNOWN_FUNCTION_-32764` appears to already be properly defined in the library. From disassembling the assembly, one can see that its id is 0x8004 and it takes two parameters, as `CAMP_FUNCTION_0004` does. It appears to be interpreting a number that should be unsigned as signed, as -32764 in hex is -7ffc, the two's complement of which is 8004.

**Comment by tge-was-taken** [comment]
Fixed by 2578a3c and 60c49e9
